### PR TITLE
sage: 9.3rc4 -> 9.3

### PR DESCRIPTION
**Pull request by collares**
###### Motivation for this change

Also updated gap and pari while at it. Pari required [an unmerged patch from Sage Trac](https://trac.sagemath.org/ticket/30801), but the patch hasn't changed in two weeks so that should be fine. Let me know if I should remove it, though.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/gap/default.nix
Comment (collares) [reply pkgs/applications/science/math/gap/default.nix:90]: Done, thanks!

**Review by timokau — CHANGES_REQUESTED** [review]
Thanks for following up on this. I would prefer to delay the pari update until the sage ticket has been reviewed. It looks like this update is likely to cause trouble.

The other changes look good.

**Comment by collares** [comment]
That's https://trac.sagemath.org/ticket/31443. I will split the pari update to a separate PR.

File: pkgs/applications/science/math/sage/sage-src.nix
Comment (collares) [reply pkgs/applications/science/math/sage/sage-src.nix]: I think all patches, once merged to Sage main, have `origin/develop` as their immediate parent, and in the process of merging such conflicts get resovled. Therefore, the problem described above only affects unmerged Trac branches. `sagemath/sage` does not contain trac branches, so we wouldn't see any change in using GitHub unless we also migrated to `sagemath/sagetrac-mirror`. But using `sagemath/sagetrac-mirror` would be impossible because the tags aren't synchronized properly there (the most recent tag now is `9.3beta2`).

Let me know if I should still switch to GitHub given that, I might be missing something.

Comment (collares) [reply pkgs/applications/science/math/sage/sage-src.nix]: To be clear, what GitHub gives us is the ability of choosing between `compare/commit1..commit2.patch`, which returns a "non-squashed" version of the commits that happened from `common_ancestor(commit1, commit2)` to `commit2` and `compare/commit..commit2.diff`, which is just a diff between the two branches. `fetchSageDiff` only offers the first possibility.

**Comment by timokau** [comment]
> That's https://trac.sagemath.org/ticket/31443.

Ah, I forgot about [this conversation](https://github.com/NixOS/nixpkgs/pull/116365#discussion_r623428696). That update sounds less risky than the pari update. I think there also isn't much urgency to it though, so I would probably wait a few days for updates on the sage ticket. Whichever way you prefer.

> I have split the pari update to #122624.

Thank you. Should be good to go now.

**Review by timokau — APPROVED** [review]
Looks good, pending ofBorg's approval.

File: pkgs/applications/science/math/sage/sage-src.nix
Comment (timokau) [reply pkgs/applications/science/math/sage/sage-src.nix]: > But using `sagemath/sagetrac-mirror` would be undesirable because the tags aren't synchronized properly there (the most recent tag now is `9.3beta2`).

You are right, that would be inconvenient. I didn't know that. Thanks for clarifying.

**Comment by collares** [comment]
@timokau Thank you for the review! I will have an eclib PR soon.

**Review by AndersonTorres — APPROVED** [review]
LGTBorg

**Pull request merged**
